docs/DeveloperGuide.md

```diff
@@ -1,34 +1,270 @@
+
+#### <a id="overview"><ins>2.1 Architecture Overview</ins></a>
+
+<p align="center">
+  <img alt="NotUS Architecture Overview" src="diagrams/out/Architecture_Overview.png" />
+  <br><em>Figure #</em>
+</p>
+
+The architecture design is given in the diagram above. The main components of NotUS are:
+
+1. `InterfaceManager`: Manages the user input as well as the message output from application.
+1. `ParserManager`: Creates a suitable parser, based on the command, to make sense of user message. The respective parsers then make sense of the information and calls the respective commands.
+1. `Command`: Executes the necessary tasks, depending on the respective command calls .
+1. `TagManager`: Stores and manages the creation and deletion of tags and other tag-related functionality.
+1. `Timetable`: Stores and manages the creation and deletion of events and other event-related functionality.
+1. `Notebook`: Stores and manages the creation and deletion of notes and other note-related functionality.
+1. `StorageManager`: Manages the loading of existing saved files and exporting of data to human-editable files
+
+A Program Evaluation Review Technique (PERT) Chart was created prior to the start of developing NotUS and was constantly updated based on progress and updates from the development team's weekly meetings. A PERT chart is a project management tool that provides a visual representation of a project's timeline. The chart breaks down the individual tasks and aids in identifying task dependencies. A diagram of the PERT chart used for this application is shown below.
+
+<p align="center">
+  <img alt="PERT Chart" src="diagrams/out/PERT_Chart.png" />
+  <br><em>Figure #</em>
+</p>
+
+#### <a id="notus"><ins>2.2 NotUS</ins></a>
+
+NotUs manages the flow of the application. On launch, it will create the necessary components, as listed above and then attempts to load any existing saved files into the application. Subsequently, it will accept and interpret the user input and execute the commands accordingly. The diagram below depicts the main flow of the application.
+
+<p align="center">
+  <img alt="NotUS" src="diagrams/out/Notus.png" />
+  <br><em>Figure #</em>
+</p>
+
+#### <a id="parserManager"><ins>2.3 ParserManager</ins></a>
+
+The ParserManager manages the creation of specific parser objects based on the type of command. The parser then makes sense of the user input and calls the respective commands into action.
+
+1. Receives the user input message as a whole.
+1. Interprets the type of command and creates the respective parser for each command.
+1. The parser then splits the message to identify all the parameters provided.
+1. Creates and returns the Command class respectively.
+ 
+<p align="center">
+  <img alt="Parser" src="diagrams/out/Parser.png" />
```

Comment:
![image](https://user-images.githubusercontent.com/60461477/97396751-ac959a00-1922-11eb-8a8b-01bd3a27e76d.png)
Under [add-n], maybe consider increasing the length of the activation bar of Command so that the return line does not overlap with the deleted ParseAddNoteCommand? Just for clarity :)

Reply:
There is a similar occurence in section 2.4, in the first diagram (see bottom of inserted pic)
![image](https://user-images.githubusercontent.com/60461477/97397367-0c407500-1924-11eb-8a61-0ee1c1ea1de3.png)